node/impl/full/chain.go

```diff
@@ -665,18 +665,41 @@ func (a *ChainAPI) ChainGetEvents(ctx context.Context, root cid.Cid) ([]types.Ev
 	if err != nil {
 		return nil, xerrors.Errorf("load events amt: %w", err)
 	}
+	if evtArr.Len() == 0 {
+		return nil, nil
+	}
+
+	// Hyperspace nv20: we peek into the events array to figure out if these
```

Comment:
This is highlighting an issue with this API. I'm not sure what the right fix is, but this is going to be painful going forward if/when we change the events structure.